### PR TITLE
Adds support for badges

### LINKED ISSUE #5366: Honey: Support chrome.browserAction.setBadgeText
## Test Plan:
1. Enable the LastPass extension
2. From chrome-extension://hdokiejnpimakedhajhdlcegeplioahd/tabDialog.html?dialog=createAccountSimple, open Developer Tools (Ctrl+Shift+I)
3. From console, run `chrome.browserAction.setBadgeText({text:"27"})`

## Original issue description
Honey (and other extensions) may often communicate valuable information by way of small text labels on or around their badge. Honey uses this API to display the total number of deals available for the domain.

**Screenshot:**
image

**Documentation:** https://developer.chrome.com/extensions/browserAction#method-setBadgeText

**Related Issues:** #5362

### LINKED ISSUE #5367: Honey: Support browserAction.setBadgeBackgroundColor
## Test Plan:
1. Enable the LastPass extension
2. From chrome-extension://hdokiejnpimakedhajhdlcegeplioahd/tabDialog.html?dialog=createAccountSimple, open Developer Tools (Ctrl+Shift+I)
3. From console, run `chrome.browserAction.setBadgeBackgroundColor({color:'#FF0000'})`

## Original issue description
Honey (and other extensions) may often communicate valuable information by way of badge icon background colors. Honey uses this API to indicate the presence of deals available for the domain.

**Screenshot:** (Left: Chrome, Right: Brave)
image

**Documentation:** https://developer.chrome.com/extensions/browserAction#method-setBadgeBackgroundColor

**Related Issues:** #5366, #5362

### PR DESCRIPTION
**Pull request by jonathansampson**
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).

Fixes #5366
Fixes #5367

## Test Plan:
1. Enable the LastPass extension
2. From chrome-extension://hdokiejnpimakedhajhdlcegeplioahd/tabDialog.html?dialog=createAccountSimple, open Developer Tools (Ctrl+Shift+I)
3. From console, run `chrome.browserAction.setBadgeText({text:"27"})`
4. From console, run `chrome.browserAction.setBadgeBackgroundColor({color:'#FF0000'})`

badge

### CODE REVIEW

**Comment by bbondy** [comment]
Super pumped you did this great work but I have to bump to 0.13.3 because we're past frozen date, sorry about that.

**Comment by jonathansampson** [comment]
@bbondy No worries; with that added time, I'll do a bit of polishing.

**Comment by bbondy** [comment]
cool thanks, 0.13.3 freeze is Monday end of day, but historically we're bad about respecting the freeze dates :)

**Comment by jonathansampson** [comment]
@bbondy I'll have my changes in before then :)

**Comment by jonathansampson** [comment]
@bradleyrichter I put in some logic to center _wider_ badges. Presently, this means anything wider than 20 pixels. Please feel free to polish up any of the styles.

layouts

**Comment by bradleyrichter** [comment]
@jonathansampson Ok, I found a tiny size that is still legible at 72dpi.

5pt Arial

padding: 1px/2px

Your centering plan should work fine.

When we apply this to the brave logo, we should use a dark grey BG color:

image

**Comment by bsclifton** [comment]
@jonathansampson I don't recall seeing an issue for this (cc: @bradleyrichter) but we have an Asana issue captured for a similar badge on the Lion icon too (so that you know how many items blocked).  Would you be able to work on that too?

**Comment by bridiver** [comment]
Badges? We don't need no stinkin' badges!

**Comment by bridiver** [comment]
++ as long as this meets @bradleyrichter's ui specs on all platforms

**Comment by alexwykoff** [comment]
screen shot 2017-02-21 at 5 07 33 pm
screen shot 2017-02-21 at 5 07 09 pm

Badges are happening!

Didn't notice any for 1pass even though I had logins for the sites, but may be due to the signing thing for local runs vs builds.

My only critique is that I find the value hard to read, but I do recognize that something is there.

My only negative mark is that the pocket icon seems to have regressed. 

Will follow up in a moment with test run results from merging latest master. No merge conflicts detected.

**Comment by alexwykoff** [comment]
1179 tests passed, 39 pending, 6 failed.
All 6 were timeouts and appeared to be of our intermittent variety.

👍 for merge pending review of pocket icon regression ^^

**Comment by bridiver** [comment]
@alexwykoff I don't think 1pw has badges. I've never seen any on Chrome or Safari

**Comment by bsclifton** [comment]
@alexwykoff are you sure the icon for Pocket regressed? When I run master on Windows (without this patch), I'm seeing that icon too.  My guess is that the extension has a new icon (maybe Windows only?)  It should be the same in Chrome on Windows

screen shot 2017-02-23 at 12 23 33 am

cc: @srirambv

**Comment by srirambv** [comment]
@bsclifton  I see it on preview 2 but only on x64 but not ia32 build.
image
image

**Comment by srirambv** [comment]
It seems to be only in Preview 2. Preview 1 has the same icon on both x64 & ia32

**Comment by bsclifton** [comment]
@jonathansampson could you squash this up and add some testing steps? @alexwykoff has already reviewed, I can check it out too and if 👍  we can merge 0.13.5

**Comment by bsclifton** [comment]
Rebased and squashed; @NejcZdovc, I added you as a review for when you have time 😄

**Comment by jonathansampson** [comment]
@bsclifton I can edit, and add a test plan. What relevance is the update action? To test this, we'd want to install an extension, and confirm that the browser action icon is one from the `manifest.browser_action.icons` property (whether it be a string, or object).

**Comment by bsclifton** [comment]
@jonathansampson sorry- I just meant update as in something that causes the numbers to show inside the icon (like in your screenshot at the top, LastPass has the number 27 in it).

I guess that is a good test- just pick LastPass as the default password manager and add some passwords 😄

File: app/renderer/components/browserAction.js [hunk 1]
```diff
@@ -34,19 +35,27 @@ class BrowserActionButton extends ImmutableComponent {
...
-        backgroundPosition: 'center'
-      }}
-      dataButtonValue={this.props.extensionId}
-      onClick={this.onClick} />
+    return <div className={cx({browserActionButton: true})}>
+      <Button iconClass='extensionBrowserAction'
+        className={cx({
+          extensionButton: true
+        })}
+        inlineStyles={{
```
Comment (NejcZdovc) [inline app/renderer/components/browserAction.js]: Let's switch to aphrodite

File: app/renderer/components/browserActionBadge.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  }
+
+  componentDidUpdate () {
+    this.determineLayout()
+  }
+
+  render () {
+    return <div
+      ref='badge'
+      className='browserActionBadge'
```
Comment (NejcZdovc) [inline app/renderer/components/browserActionBadge.js]: Let's use aphrodite

File: less/navigationBar.less [hunk 3]
```diff
@@ -511,11 +511,37 @@
     flex-direction: row;
     margin-left: 3px;
 
+    .browserActionButton {
```
Comment (NejcZdovc) [inline less/navigationBar.less]: No more less :smiley:

**Comment by bsclifton** [comment]
@jonathansampson we can help you with the Aphrodite refactor (in fact, @cezaraugusto had [put together a really sweet guide](https://github.com/brave/browser-laptop/wiki/Refactoring-styles-to-Aphrodite) which you might find helpful 😄 )

In the meantime, I'll move this to 0.13.7 😄

**Comment by NejcZdovc** [comment]
@bsclifton @jonathansampson yeah I can help you with it, not a problem

**Comment by jonathansampson** [comment]
@bsclifton Sorry for the _Testing Plan_ confusion. Half-way through this thread became about a Pocket icon regression. That threw me off.

Test Plan is now added. Essentially, call the two primary APIs from a LastPass extension resource. When enabled, LastPass opens up a chrome-extension: resource for the purposes of creating an account, or logging in. This page will suffice (as far as access/privileges go) to update the LastPass badge with color and text.

**Comment by jonathansampson** [comment]
@NejcZdovc Moved to Aphrodite (as much as I could). There are a couple of places where we don't know the value of the style property until runtime. Let me know if there's anything that could be further improved.

**Comment by jonathansampson** [comment]
@NejcZdovc @bridiver @bradleyrichter Can I get a fresh review? :)

File: app/renderer/components/browserActionBadge.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      )}
+      style={{backgroundColor: this.props.color || commonStyles.color.braveMediumOrange}}
+      >{this.props.text}</div>
+  }
+}
+
+const styles = StyleSheet.create({
+  browserActionBadge: {
+    color: 'white',
+    bordeRadius: '2px',
```
Comment (NejcZdovc) [inline app/renderer/components/browserActionBadge.js]: Typo: `bordeRadius` -> `borderRadius`

File: app/renderer/components/browserActionBadge.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        this.centered && styles.centered
+      )}
+      style={{backgroundColor: this.props.color || commonStyles.color.braveMediumOrange}}
+      >{this.props.text}</div>
+  }
+}
+
+const styles = StyleSheet.create({
+  browserActionBadge: {
+    color: 'white',
```
Comment (NejcZdovc) [inline app/renderer/components/browserActionBadge.js]: Please use hex `#FFF`

**Comment by bridiver** [comment]
can we get some tests for this? Preferably with a mock extension that is only enabled for testing

File: app/renderer/components/browserActionBadge.js (hunk 5 above)
Comment (jonathansampson) [inline app/renderer/components/browserActionBadge.js]: Done.

File: app/renderer/components/browserActionBadge.js (hunk 4 above)
Comment (jonathansampson) [inline app/renderer/components/browserActionBadge.js]: Done.

**Comment by jonathansampson** [comment]
@bridiver I believe that's doable.

**Pull request merged**